tests/test_stimulation_signal.py

```diff
@@ -1,179 +1,226 @@
-# TODO implement meaningful tests for signals
-"""
-from ossdbs.stimulation_signals.rectangle_signal import RectangleSignal
-from ossdbs.stimulation_signals.triangle_signal import TriangleSignal
-from ossdbs.stimulation_signals.trapezoid_signal import TrapezoidSignal
 import numpy as np
 import pytest
 
+from ossdbs.stimulation_signals.rectangle_signal import RectangleSignal
+from ossdbs.stimulation_signals.trapezoid_signal import TrapezoidSignal
+from ossdbs.stimulation_signals.triangle_signal import TriangleSignal
+
 
 class TestRectangleStimulationSignal:
-
     def test_generate_samples(self):
-        signal = RectangleSignal(frequency=1,
-                                 pulse_width=0.3,
-                                 counter_pulse_width=0.6,
-                                 inter_pulse_width=0.1)
-        actual = signal._generate_samples(sample_spacing=0.1)
-        desired = [1, 1, 1, 0, -0.5, -0.5, -0.5, -0.5, -0.5, -0.5]
+        signal = RectangleSignal(
+            frequency=1,
+            pulse_width=0.3,
+            counter_pulse_width=0.5,
+            inter_pulse_width=0.1,
+            counter_pulse_amplitude=0.5,
+        )
+        actual = signal.get_time_domain_signal(dt=0.1, timesteps=12)
+        desired = [0, 0, 0, 1, 1, 1, 0, -0.5, -0.5, -0.5, -0.5, -0.5]
```

Comment:
Here, the offset could actually be a problem. We shifted the signals because it is nicer to visualize.